schematic/models/metadata.py

```diff
@@ -235,3 +237,45 @@ def populateModelManifest(self, title, manifestPath: str, rootNode: str) -> str:
         emptyManifestURL = mg.get_manifest()
 
         return mg.populate_manifest_spreadsheet(manifestPath, emptyManifestURL)
+
+
+    def submit_metadata_manifest(self, manifest_path: str, dataset_id: str, validate_component: str, json_schema: str = None) -> bool:
+        """Wrap methods that are responsible for validation of manifests for a given component, and association of the 
+        same manifest file with a specified dataset.
```

Comment:
If if validate_component is None and the jsonschema is provided, then validate using the provided jsonschema.

Reply:
My thoughts on this are that if `validate_component` is not specified, then the user doesn't want us to validate the manifest at all, which would mean no JSON Schema generation (and validation). So I thought it would be a good idea to handle it like on [L#300](https://github.com/Sage-Bionetworks/schematic/blob/develop-submit_metadata_manifest/schematic/models/metadata.py#L300). If we still want to do JSON Schema validation I can change the code.

Reply:
I was thinking that we can provide the option for the user to give us their own jsonschema (e.g. could be outside of the data model) and running the validator on the related manifest. But I agree this is a use case that can be deprioritized, especially given development in Synapse supporting jsconschema validation. So, I'd agree with your approach; in this case do we still even want to expose the jsonschema as an argument option (i.e. only the validate_component is relevant)?

Reply:
Right, so the `validateModelManifest()` method expects both `rootNode` (`validate_component`) and (`jsonSchema) `json_schema`. So if we want to provide the option of validating without `rootNode`, I'll need to change the signature of `validateModelManifest()` and `get_json_schema_requirements()` (which is not too bad). But you mentioned that use-case can be de-prioritized, so we'll ignore that for now.

So the one case-scenario in which exposing the `json_schema` argument becomes useful is when the user provides a `rootNode` / `validate_component`, but wants custom JSON Schema validation rules. Do we want that flexibility? If not, I can deprecate the use of `json_schema`.

Reply:
Yes, we don't want to support scenarios (at least at present) where the user provides a rootNode but provides custom json_schema; so let's go ahead and deprecate the use of json_schema. If needed we can introduce the option in validate_manifest later.

Reply:
The `json_schema` argument has been deprecated in this [commit](https://github.com/Sage-Bionetworks/schematic/pull/353/commits/7aa1fd021a194a767f77271baa68d61cceaf4524).